akka-typed/src/main/scala/akka/typed/cluster/ClusterSingleton.scala

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster
+
+import akka.actor.{ ExtendedActorSystem, InvalidActorNameException, NoSerializationVerificationNeeded }
+import akka.annotation.{ DoNotInherit, InternalApi }
+import akka.cluster.ClusterSettings.DataCenter
+import akka.cluster.singleton.{ ClusterSingletonManagerSettings, ClusterSingletonProxy, ClusterSingletonProxySettings, ClusterSingletonManager ⇒ OldSingletonManager }
+import akka.typed.internal.adapter.{ ActorRefAdapter, ActorSystemAdapter }
+import akka.typed.scaladsl.adapter._
+import akka.typed.{ ActorRef, ActorSystem, Behavior, Extension, ExtensionId, Props }
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.FiniteDuration
+
+object ClusterSingletonSettings {
+  def apply[A](
+    system: ActorSystem[_]
+  ): ClusterSingletonSettings = {
+    // TODO read existing cluster singleton settings or introduce its own config namespace?
+    // currently singleton name is required and then discarded
+    val mgrSettings = ClusterSingletonManagerSettings(system.settings.config.getConfig("akka.cluster.singleton"))
+    val proxySettings = ClusterSingletonProxySettings(system.settings.config.getConfig("akka.cluster.singleton-proxy"))
+    new ClusterSingletonSettings(
+      mgrSettings.role,
+      proxySettings.dataCenter,
+      proxySettings.singletonIdentificationInterval,
+      mgrSettings.removalMargin,
+      mgrSettings.handOverRetryInterval,
+      proxySettings.bufferSize
+    )
+  }
+
+  def fromConfig[A](
```

Comment:
note the not used A, though I guess you'll notice; same in apply

Reply:
Yeah, had more stuff in the settings in one iteration left over from that. I'll remove